paperqa/types.py

```diff
@@ -570,17 +576,19 @@ def formatted_citation(self) -> str:
                 " to call `hydrate`?"
             )
 
-        quality = (
-            SOURCE_QUALITY_MESSAGES[self.source_quality]
-            if self.source_quality >= 0
-            else None
-        )
+        if self.source_quality_str:
+            return (
+                f"{self.citation} This article has {self.citation_count} citations and is"
+                f" from a {self.source_quality_str}."
+            )
+        return f"{self.citation} This article has {self.citation_count} citations."
 
-        if quality is None:
-            return f"{self.citation} This article has {self.citation_count} citations."
+    @property
+    def source_quality_str(self) -> str:
```

Comment:
```suggestion
    def source_quality_message(self) -> str:
```

Let's not restate the return type in the name, it's redundant